src/uart/mod.rs

```diff
@@ -0,0 +1,73 @@
+//! Using the UART
+use crate::error::{NegativeErrorExt, NumericError};
+use embedded_io;
+use embedded_io::ErrorKind;
+use riot_sys::libc::c_void;
+use riot_sys::uart_init;
+use riot_sys::uart_t;
+use riot_sys::uart_write;
+
+#[derive(Debug)]
+pub struct UARTDevice {
+    dev: uart_t,
+}
+
+impl embedded_io::Error for NumericError {
+    fn kind(&self) -> ErrorKind {
+        match -self.number() as _ {
+            riot_sys::ENODEV => ErrorKind::NotFound,
+            riot_sys::ENOTSUP => ErrorKind::Unsupported,
+            _ => ErrorKind::Other,
+        }
+    }
+}
+
+impl embedded_io::ErrorType for UARTDevice {
+    type Error = NumericError;
+}
+
+impl UARTDevice {
+    /// Create a new UARTDevice from a RIOT descriptor
+    pub fn from_c(dev: uart_t) -> Self {
+        UARTDevice { dev }
+    }
+
+    /// Create a new UARTDevice from device number
+    pub fn from_port(dev: u8) -> Self {
```

Comment:
```suggestion
    pub fn from_port(dev: u32) -> Self {
        UARTDevice { riot_sys::macro_UART_DEV(dev) }
```

(The assumption that c's int is u32 is widespread in RIOT and the wrappers; if that were ever challenged, that'll require breaking changes in several places)